scripts/plugin-directory.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+set -euo pipefail
+
+OUTPUT_DIR="target/wordpress-org-plugin-directory"
+
+fetch_plugin_data() {
+    slug=$1
+    curl -s --output "$OUTPUT_DIR/$slug.json" "$API_URL?action=plugin_information&fields=icons%2Cbanners&slug=$slug"
+}
+
+download_from_wp_org() {
+    echo "Downloading plugin data from WordPress.org."
+    echo "(Run this script within sandbox to get a much faster download speed.)"
+
+    rm -rf "$OUTPUT_DIR" && mkdir -p "$OUTPUT_DIR"
+
+    API_URL="https://api.wordpress.org/plugins/info/1.2/"
+    PARALLEL_JOBS=10
+
+    current_page=1
+
+    echo "Fetching the total number of pages..."
+    response=$(curl -s "$API_URL?action=query_plugins&request%5Bpage%5D=$current_page&request%5Bper_page%5D=100")
+    total_pages=$(echo "$response" | jq -r '.info.pages')
+    if [[ -z "$total_pages" || "$total_pages" -eq 0 ]]; then
+        echo "Failed to fetch pagination info or no pages available."
+        exit 1
+    fi
+
+    echo "Total pages to process: $total_pages"
+
+    export -f fetch_plugin_data
+    export OUTPUT_DIR
+    export API_URL
+
+    while [[ $current_page -le $total_pages ]]; do
+        echo "[$(date)] Processing page: $current_page"
+
+        response=$(curl -s "$API_URL?action=query_plugins&request%5Bpage%5D=$current_page&request%5Bper_page%5D=100")
+        plugin_slugs=$(echo "$response" | jq -r '.plugins[].slug')
+
+        echo "$plugin_slugs" | xargs -n 1 -P "$PARALLEL_JOBS" bash -c 'fetch_plugin_data "$@"' _
+
+        current_page=$((current_page + 1))
+    done
+
+    echo "All plugin data has been saved to the directory: $OUTPUT_DIR"
+}
+
+S3_URI="s3://a8c-ci-cache/wordpress-rs-wordpress-org-plugin-directory.tar.gz"
+
+upload_to_s3() {
+    echo "Uploading $(ls -1 $OUTPUT_DIR | wc -l) plugins in $OUTPUT_DIR to S3..."
```

Comment:
[Shellcheck](https://www.shellcheck.net/) says:

```
echo "Uploading $(ls -1 $OUTPUT_DIR | wc -l) plugins in $OUTPUT_DIR to S3..."
                  ^-- [SC2012](https://www.shellcheck.net/wiki/SC2012) (info): Use find instead of ls to better handle non-alphanumeric filenames.
                        ^-- [SC2086](https://www.shellcheck.net/wiki/SC2086) (info): Double quote to prevent globbing and word splitting.
```

Reply:
Related to my suggestion below – I think this entire script would be replaced by a single Rust test instead?